hack/generator/pkg/test/asserts.go

```diff
@@ -28,3 +29,27 @@ func AssertFileGeneratesExpectedCode(t *testing.T, fileDef *astmodel.FileDefinit
 
 	g.Assert(t, testName, buf.Bytes())
 }
+
+// AssertPackagesGenerateExpectedCode creates a golden file for each package represented in the passed set of type
+// definitions, asserting that the generated content is expected
+func AssertPackagesGenerateExpectedCode(t *testing.T, types astmodel.Types, prefix string) {
```

Comment:
It'd be nice to have the files each in a subfolder with the name of the test? Otherwise feels like we'll get just a crazy number of golden files and it'll be hard to tell which apply to what? I'm actually struggling with that in this PR even with the tests you've added.

Reply:
Agreed. See #1626